### DISCUSSION TITLE
What is --resource flag?

### DISCUSSION THREAD

**Discussion by Blank-c** [pyinstaller/pyinstaller#7726] [PyInstaller] [answered]
I tried to do `pyinstaller --resource dummy --onefile main.py`. The `dummy` file was 50 MB, but the output file was only 7 mb.

**Comment by rokm** [comment]
Based on [documentation](https://pyinstaller.org/en/stable/usage.html#cmdoption-r), if you do not specify resource name and type, the file should be an exe/dll (in which case all its resources are copied over). So if `dummy` was not an exe/dll, it was probably skipped, and you should see [this message](https://github.com/pyinstaller/pyinstaller/blob/63738a41d43839fa905473217e0141a00053c7e3/PyInstaller/building/api.py#L731) in the build log.

**Comment by bwoodsend** [comment]
`--add-data` is probably the flag you're really looking for.

**Reply by Blank-c** [reply]
I know but I just want to test the `--resource` flag. How do we use it?

**Reply by bwoodsend** [reply]
I actually have no idea what it's for.

**Reply by rokm** [reply, marked as answer]
It's for embedding [resources](https://learn.microsoft.com/en-us/windows/win32/menurc/resources) in executable, for use with win32 resource API (e.g., [FindResourceW](https://learn.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-findresourcew)).

**Reply by Blank-c** [reply]
Whats the syntax, like I did `pyinstaller --resource dummy --onefile main.py`, dummy was an exe file likely 42 mb and it was not embeded.

**Reply by rokm** [reply]
> Whats the syntax, like I did `pyinstaller --resource dummy --onefile main.py`, dummy was an exe file likely 42 mb and it was not embeded.

If it was executable and you provided just filename, it was copying *resources* from that executable, not embedding the executable itself.

See the docs: https://pyinstaller.org/en/stable/usage.html#cmdoption-r

**Comment by Blank-c** [comment]
> > Whats the syntax, like I did `pyinstaller --resource dummy --onefile main.py`, dummy was an exe file likely 42 mb and it was not embeded.
> 
> If it was executable and you provided just filename, it was copying _resources_ from that executable, not embedding the executable itself.
> 
> See the docs: https://pyinstaller.org/en/stable/usage.html#cmdoption-r

Ah okay. Also do you know how can I programically find the start position of the embed Archive in Windows 10 pyinstaller?

**Reply by rokm** [reply]
> Ah okay. Also do you know how can I programically find the start position of the embed Archive in Windows 10 pyinstaller?

You mean PyInstaller's PKG/CArchive? Check out [the source code for `CArchiveReader`](https://github.com/pyinstaller/pyinstaller/blob/63738a41d43839fa905473217e0141a00053c7e3/PyInstaller/archive/readers.py#L87-L100)...
